src/Chocolatey.PowerShell/Helpers/ProcessInformation.cs

```diff
@@ -0,0 +1,49 @@
+﻿// Copyright © 2017 - 2024 Chocolatey Software, Inc
+// Copyright © 2011 - 2017 RealDimensions Software, LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+//
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System.Runtime.InteropServices;
+using System.Security.Principal;
+
+namespace Chocolatey.PowerShell.Helpers
+{
+    public static class ProcessInformation
+    {
+        /// <summary>
+        /// Helper method for determining current OS platform.
+        /// </summary>
+        /// <returns>True if the current OS is Windows.</returns>
+        public static bool IsWindows()
+        {
+            return RuntimeInformation.IsOSPlatform(OSPlatform.Windows);
+        }
+
+        /// <summary>
+        /// Determines whether the current process has administrative rights.
+        /// </summary>
+        /// <returns>True if running on Windows and the process has administrative rights.</returns>
+        public static bool HasAdminRights()
+        {
+            if (!IsWindows())
+            {
+                return false;
+            }
+
+            var currentPrincipal = new WindowsPrincipal(WindowsIdentity.GetCurrent(TokenAccessLevels.Query | TokenAccessLevels.Duplicate));
+
+            return currentPrincipal.IsInRole(WindowsBuiltInRole.Administrator);

```

Comment:
I'm not sure I see a reason _not_ to fix that here tbh, if we have a clear idea of what this should return if the current identity returns null? Should it just return false?

Reply:
The reason I said to create a followup issue is based on what @gep13 mentioned to me that the code is meant to be basically a copy of the PowerShell code. Since it is meant to be basically a copy, and the PowerShell code didn't already have these check, it probably means they won't happen that often.

I'm not against fixing it now while we have the chance, though.

Reply:
> Should it just return false?

Returning false sounds appropriate to me, that is what the original handler in Chocolatey CLI does

```cs
if (Platform.GetPlatform() != PlatformType.Windows) 
{ 
    return false; 
} 

using (var identity = WindowsIdentity.GetCurrent(TokenAccessLevels.Query | TokenAccessLevels.Duplicate)) 
{ 
    if (identity != null) 
    { 
        var principal = new WindowsPrincipal(identity); 
        return principal.IsInRole(WindowsBuiltInRole.Administrator); 
     } 
 }
 
return false; 
```

Reply:
I'd prefer to fix it here, just because I _think_ PowerShell has some behind-the-scenes nonsense that allows it to properly dispose things that fall out of scope even if users forget, and we may not have the same in C# (although the destructor should run _eventually_ once the GC finds it).

Reply:
I'll push this fix shortly.